incubator/kafka/templates/statefulset.yaml

```diff
@@ -51,7 +51,7 @@ spec:
         command:
         - sh
         - -c
-        - "./bin/kafka-server-start.sh config/server.properties --override zookeeper.connect={{ .Release.Name }}-zookeeper:2181/ --override broker.id=${HOSTNAME##*-}"
+        - "./bin/kafka-server-start.sh config/server.properties --override zookeeper.connect={{ .Release.Name }}-zookeeper:2181/ --override log.dirs={{ printf "%s/logs" .Values.DataDirectory | trunc 63 }} --override broker.id=${HOSTNAME##*-}"
```

Comment:
I don't want to move Kafka configs to config-map because of this single change.
It is common to tweak some of its config values and it would be good to have them editable, but maybe this should done as a separate task?

This change only restore previous valid behavior that got broken in one of last PRs. Please see commit message for links.